lib/devise/multi_email/models/confirmable.rb

```diff
@@ -38,64 +38,108 @@ module ConfirmableExtensions
 
         included do
           multi_email_association.include_module(EmailConfirmable)
-        end
-
-        # delegate before creating overriding methods
-        delegate :skip_confirmation!, :skip_confirmation_notification!, :skip_reconfirmation!, :confirmation_required?,
-                 :confirmation_token, :confirmed_at, :confirmation_sent_at, :confirm, :confirmed?, :unconfirmed_email,
-                 :reconfirmation_required?, :pending_reconfirmation?, to: :primary_email_record, allow_nil: true
-
-        # This need to be forwarded to the email that the user logged in with
-        def active_for_authentication?
-          login_email = current_login_email_record
 
-          if login_email && !login_email.primary?
-            super && login_email.active_for_authentication?
-          else
-            super
+          # Handle automatically confirming and linking `unconfirmed_email` if email change is not postponed
+          before_update do
```

Comment:
Devise has the concept of "postpone email change", which means it saves the changed email in the `unconfirmed_email` column. If for some reason Devise settings are such to allow users to change their email without confirming first, that is the scenario that this handles. It is likely never the case if you're using this gem, but it's here to handle the Devise lifecycle expectations.